core/src/main/java/org/apache/iceberg/BaseSnapshot.java

```diff
@@ -172,15 +170,7 @@ private void cacheManifests(FileIO fileIO) {
 
     if (allManifests == null) {
       // if manifests isn't set, then the snapshotFile is set and should be read to get the list
-      InputFile manifestListInputFile;
-
-      if (manifestListFile.encryptedKeyMetadata() == null) {
-        manifestListInputFile = fileIO.newInputFile(manifestListFile.location());
-      } else { // encrypted manifest list file
-        manifestListInputFile = EncryptionUtil.decryptManifestListFile(manifestListFile, fileIO);
-      }
-
-      this.allManifests = ManifestLists.read(manifestListInputFile);
+      this.allManifests = ManifestLists.read(fileIO.newInputFile(manifestListFile));
```

Comment:
As with the other file types, `FileIO` and `EncryptingFileIO` are now responsible for opening the input file rather than having the caller handle encryption.